### PR TITLE
ETSWORK-10471 - Convert `@testing-library/react` and `JSDOM` tests to use Cypress component tester

### PR DESCRIPTION
**Pull request by coreyjv**
There were a number of reasons for the conversion to Cypress component tester but the two major reasons were:

1. Ionic is based on Web Components and `JSDOM` has very limited support
2. Both `@testing-library/react` amd Ionic must fake events which reduces confidence that components will work in practice.

I decided to go with a `.cypress.tsx` syntax so we can isolate Cypress tests from the other tests such as image snapshots and A11y tests. This will also future proof us if we ever add plain old unit tests that can still run with Jest.

The build has been updated accordingly with an additional step to ensure Cypress tests pass with screenshots and videos being uploaded as artifacts for failures.

Thanks to @TuckerWhitehouse for helping get the base Cypress setup in place.

### CODE REVIEW

**Review by stevetothdesign — APPROVED** [review]

File: cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
Comment (TuckerWhitehouse) [inline cypress/fixtures/example.json:5]: Not that it matters but could probably delete this boilerplate file

File: tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "include": ["src/**/*"],
-  "exclude": ["node_modules", "**/*.test.tsx", "**/*.stories.tsx", "src/storyshots"],
+  "exclude": ["node_modules", "**/*.test.tsx", "**/*.stories.tsx", "**/*.cypress.tsx", "src/storyshots"],
```
Comment (TuckerWhitehouse) [inline tsconfig.json:3]: I wonder if this is why my editor was being annoying -- may need a tsconfig that doesn't exclude test files, and one that does 🤔

**Review by TuckerWhitehouse — APPROVED** [review]

File: src/components/NetworkSplash.cypress.tsx [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
+    mount(
       <NetworkSplash isNetworkAvailable={true} translations={translations}>
         <p>{`${childComponentText}`}</p>
       </NetworkSplash>
     )
 
-    expect(screen.getByText(childComponentText)).toBeVisible()
-    expect(screen.queryByText(translations.cantConnectToNetwork)).not.toBeInTheDocument()
+    cy.findByText(childComponentText).should('exist')
+    cy.findByText(translations.cantConnectToNetwork).should('not.exist')
```
Comment (louise-hayes) [inline src/components/NetworkSplash.cypress.tsx:21]: This is so much easier to read not that I am biased to cypress or anything :-) Great job Guys getting this up and running!

Comment (louise-hayes) [inline src/components/NetworkSplash.cypress.tsx:21]: This is so much easier to read not that I am biased to cypress or anything :-) Great job Guys getting this up and running!

**Review by louise-hayes — APPROVED** [review]

File: cypress/fixtures/example.json (hunk 1 above)
Comment (coreyjv) [reply cypress/fixtures/example.json:5]: I agree. I'll get this cleaned up in my next PR.

File: tsconfig.json (hunk 2 above)
Comment (coreyjv) [reply tsconfig.json:3]: I feel like my editor hinting has been working pretty well.

**Pull request merged**